runtime/src/accounts.rs

```diff
@@ -499,36 +495,13 @@ impl Accounts {
         txs: &[Transaction],
         txs_iteration_order: Option<&[usize]>,
     ) -> Vec<Result<()>> {
-        let mut error_counters = ErrorCounters::default();
-        let keys: Vec<_> = OrderedIterator::new(txs, txs_iteration_order)
-            .map(|tx| {
-                let message = &tx.message();
-                message.get_account_keys_by_lock_type()
+        let mut account_locks = &mut self.account_locks.lock().unwrap();
+        OrderedIterator::new(txs, txs_iteration_order)
+            .map(|tx| tx.message().get_account_keys_by_lock_type())
```

Comment:
Is `get_account_keys_by_lock_type` expensive? We're holding the `account_locks` for the duration of that call now over the original implementation

Reply:
It's not expensive, but probably good practice to hold locks for as short of a duration as possible. I went back to the original implementation 👍